### PR TITLE
Added Altair GraphQL UI.

### PR DESCRIPTION
**Pull request by imolorhe**
Added [Altair GraphQL Client](https://altair.sirmuel.design/) as an alternative GraphQL UI tool.
Altair comes packed with a lot of useful features to make it easy to work with GraphQL, and would be very useful to a lot of developers.

### CODE REVIEW

File: src/Ui.Altair/AltairGraphQLOptions.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+using Microsoft.AspNetCore.Http;
+using System.Collections.Generic;
+
+namespace GraphQL.Server.Ui.Altair
+{
+    /// <summary>
+    /// Options to customize <see cref="AltairMiddleware"/>
+    /// </summary>
+    public class AltairGraphQLOptions
```
Comment (sungam3r) [inline src/Ui.Altair/AltairGraphQLOptions.cs]:
```suggestion
    public class GraphQLAltairOptions
```

File: src/Ui.Altair/AltairExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    /// <summary>
+    /// Extension methods for <see cref="IApplicationBuilder"/>
+    /// </summary>
+    public static class PlaygroundExtensions
+    {
+        /// <summary> Adds middleware for Altair GraphQL using the specified options. </summary>
+        /// <param name="app"> <see cref="IApplicationBuilder"/> to configure an application's request pipeline. </param>
+        /// <param name="options"> Options to customize <see cref="AltairMiddleware"/>. If not set, then the default values will be used. </param>
+        /// <returns> The reference to provided <paramref name="app"/> instance. </returns>
+        public static IApplicationBuilder UseAltairGraphQL(this IApplicationBuilder app, AltairGraphQLOptions options = null)
```
Comment (sungam3r) [inline src/Ui.Altair/AltairExtensions.cs]:
```suggestion
        public static IApplicationBuilder UseGraphQLAltair(this IApplicationBuilder app, GraphQLAltairOptions options = null)
```

File: src/Ui.Altair/AltairExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
+using GraphQL.Server.Ui.Altair;
+
+namespace Microsoft.AspNetCore.Builder
+{
+    /// <summary>
+    /// Extension methods for <see cref="IApplicationBuilder"/>
+    /// </summary>
+    public static class PlaygroundExtensions
```
Comment (sungam3r) [inline src/Ui.Altair/AltairExtensions.cs]:
```suggestion
    public static class AltairExtensions
```

File: src/Ui.Altair/AltairGraphQLOptions.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        /// </summary>
+        public PathString Path { get; set; } = "/ui/altair";
+
+        /// <summary>
+        /// The GraphQL EndPoint
+        /// </summary>
+        public PathString GraphQLEndPoint { get; set; } = "/graphql";
+
+        /// <summary>
+        /// The GraphQL Config
```
Comment (sungam3r) [inline src/Ui.Altair/AltairGraphQLOptions.cs]: Change this please to something more specific to Altair.

File: samples/Samples.Server/Startup.cs [hunk 5]
```diff
@@ -98,6 +99,12 @@ public void Configure(IApplicationBuilder app)
                 GraphQLEndPoint = "/graphql",
             });
 
+            app.UseAltairGraphQL(new AltairGraphQLOptions
```
Comment (sungam3r) [inline samples/Samples.Server/Startup.cs]: It would be nice to use `optons.Headers` in this example to demonstrate some features.

**Comment by sungam3r** [comment]
@imolorhe Thank you very much for your work. I just recently read about Altair and tried to use it as desktop standalone app. I am glad that it is possible to connect Altair as html page.

Note that you will have to add a new package to build scripts - https://github.com/graphql-dotnet/server/blob/develop/build.cake#L14

File: src/Ui.Altair/AltairExtensions.cs (hunk 3 above)
Comment (imolorhe) [reply src/Ui.Altair/AltairExtensions.cs]: Sorry for the oversight. It was a copy-paste job 😄

File: samples/Samples.Server/Startup.cs (hunk 5 above)
Comment (imolorhe) [reply samples/Samples.Server/Startup.cs]: What kind of headers do you think I should add? Or would a generic `X-api-token` header be sufficient for the example?

**Comment by imolorhe** [comment]
> Note that you will have to add a new package to build scripts - https://github.com/graphql-dotnet/server/blob/develop/build.cake#L14

I already added the project here: https://github.com/graphql-dotnet/server/blob/be914f9a608ccc03ab699c8cd8cb3914ef0b41bb/build.cake#L16

Do I need to do something else?

**Comment by imolorhe** [comment]
> I just recently read about Altair and tried to use it as desktop standalone app.

I'm curious as to where you read about Altair for the first time? 🙂

**Comment by imolorhe** [comment]
I should also add that Altair comes with many more configuration options: https://github.com/imolorhe/altair#configuration-options

But I figured it's okay for the first PR not to have them (lesser code, faster review). Also would appreciate if some core maintainer could add those options instead of myself.

File: samples/Samples.Server/Startup.cs (hunk 5 above)
Comment (sungam3r) [reply samples/Samples.Server/Startup.cs]: Any are suitable just to demonstrate that they can be transmitted.

**Comment by sungam3r** [comment]
> Do I need to do something else?

I overlooked it, everything is fine.

File: samples/Samples.Server/Startup.cs (hunk 5 above)
Comment (imolorhe) [reply samples/Samples.Server/Startup.cs]: I added an example. Please review.

**Comment by sungam3r** [comment]
> I'm curious as to where you read about Altair for the first time?

https://github.com/APIs-guru/graphql-over-http/issues/48

**Comment by sungam3r** [comment]
> I should also add that Altair comes with many more configuration options

You may add link to https://github.com/imolorhe/altair#configuration-options for `GraphQLAltairOptions` summary comments.

**Comment by sungam3r** [comment]
@joemcbride @pekkah New package was uploaded to myget feed: https://ci.appveyor.com/project/graphql-dotnet-ci/server/builds/30054845#L558 but requires (as I understand) manual push to nuget. Please tell me, do I need any special keys / access rights for this?

**Comment by imolorhe** [comment]
Thanks @sungam3r!

I see that there is also docs for GraphiQL here: https://graphql-dotnet.github.io/docs/getting-started/graphiql

Would it make sense to add Altair to the docs as well?

**Comment by sungam3r** [comment]
Of course.

**Comment by imolorhe** [comment]
@sungam3r Looks like the Altair package isn't still available.

**Comment by sungam3r** [comment]
I know. @joemcbride could you please publish it?

**Pull request merged**